add richestCustomerWealth page

raveenreddy0 opened the pull request:
PR to add richestCustomerWealth.tsx and linesOfCode.ts in /api/coding/ folder to store the leetcode solution in a list of objects.

Some notes:
- page currently displays the leetcode question [prompt](https://leetcode.com/problems/richest-customer-wealth/), my solution, next / previous buttons, inputs for leetcode question, and variables in solution
- idea behind next / previous buttons is to iterate through solution step by step and highlight each line while visualizing how the variables change while working through the solution
- the solution is rendered through a map of array `linesOfCode` but I can't figure out how to highlight each line depending on the state variable `lineNum`
- currently all lines of code are highlighted when `lineNum` = 6; 6 being a placeholder for where the current `lineNum` would be referenced
- forgot to create a new branch so the old js question commits are still included in this PR whoops 

Here is what it looks so far:
<img width="1368" alt="Screen Shot 2022-02-11 at 1 44 55 PM" src="https://user-images.githubusercontent.com/66029765/153651013-c5f9c7c2-34fb-46b2-8de7-31775fd2395f.png">

LavaBoy202 commented on pages/api/coding/LinesOfCode.ts:
```diff
@@ -0,0 +1,82 @@
+//     // sort list descending
+//     balances.sort((a, b) => {return b-a})
+    
+//     return balances[0] 
```
Don't commit commented out code

LavaBoy202 commented on components/coding/JSQuiz.tsx:
```diff
@@ -58,6 +58,51 @@ const JSQuiz = ({}: JSQuizProp) => {
       answer: "variable x;",
     },
+    {
+      displayQuestion: "Which of the following expressions evaluates as True",
```
It's good to note all your changes in a PR. Add a bullet point in your PR description saying that you added some more questions to the JS quiz.

LavaBoy202 reviewed (commented):
Nice job! It's pretty clean code, just couple of small improvements here and there. Make sure the vercel tests pass before merging.

vithushan19 commented on pages/api/coding/LinesOfCode.ts:
```diff
@@ -0,0 +1,82 @@
+export const linesOfCode = [
+    {
+        line: 1,
```
This can be array of strings where you use the index

vithushan19 commented on pages/coding/richestCustomerWealth.tsx:
```diff
@@ -0,0 +1,79 @@
+            Code to Evaluate
+          </p>
+
+          {linesOfCode.map(({ line, text }) => {
```
🔥

vithushan19 reviewed (commented):
Looking good!

raveenreddy0 commented:
@vithushan19 @LavaBoy202 

Realized I've never closed the original PR and I'm getting kind of lost in this branch so sorry for that - will try to close this one out and be better about that going forward. I did try to add commits as I go so hopefully that makes it easier to follow. 

The new commits here iterate / highlight through the solution more dynamically based on Lavan's suggestions 

Some highlights / issues:
- added several new state variables to keep track of the solution steps and variables as we iterate line by line
- added Reset button that sets all state variables to initial values onClick
- created an iterateOnSolution function called by Next Button's onClick which has conditional logic to iterate through both loops appropriately - i'm guessing there's a more elegant / efficient way to work through this so can work on refactoring in next iterations
- added some code to highlight each element of the Accounts array as it's iterated through 
- i'm trying to sort through the balances array once it's populated, and pass it through setBalancesState but this code doesn't appear to be working
- in general looking for a cleaner way to render the arrays.. for the balances arrays, I created a renderAccounts object bu this seems kinda unclean??
